config/config/src/index.ts

```diff
@@ -451,6 +442,10 @@ export async function getConfig (opts: {
     pnpmConfig.filterProd = (pnpmConfig.filterProd as string).split(' ')
   }
 
+  if (pnpmConfig.filter?.length > 0 || pnpmConfig.filterProd?.length > 0) {
+    delete pnpmConfig.workspacePackagePatterns
+  }
+
```

Comment:
I don't understand why this helps. Basically you are ignoring the patterns from `pnpm-workspace.yaml`?